pkg/console/operator/sync_v400.go

```diff
@@ -411,6 +412,52 @@ func (co *consoleOperator) SyncConfigMap(
 	return cm, cmChanged, "ConsoleConfigBuilder", cmErr
 }
 
+// Build telemetry configuration in following order:
+//  1. check if the telemetry client is available and set the "TELEMETER_CLIENT_DISABLED" annotation accordingly
+//  2. get CLUSTER_ID from the cluster version config
+//  3. get ORGANIZATION_ID from OCM
+func (co *consoleOperator) GetTelemeterConfiguration(ctx context.Context, operatorConfig *operatorv1.Console) (map[string]string, error) {
+	telemetryConfig := make(map[string]string)
+	telemeterClientIsAvailable, err := telemetry.IsTelemeterClientAvailable(co.monitoringDeploymentLister)
+	if err != nil {
+		return telemetryConfig, err
+	}
+
+	if !telemeterClientIsAvailable {
+		telemetryConfig["TELEMETER_CLIENT_DISABLED"] = "true"
+		return telemetryConfig, nil
+	}
+
+	clusterID, err := telemetry.GetClusterID(co.clusterVersionLister)
+	if err != nil {
+		return nil, err
+	}
+	telemetryConfig["CLUSTER_ID"] = clusterID
+	accessToken, err := telemetry.GetAccessToken(co.configNSSecretLister)
+	if err != nil {
+		return nil, err
+	}
+
+	organizationID, err := telemetry.GetOrganizationID(clusterID, accessToken)
+	if err != nil {
+		return nil, err
+	}
+	telemetryConfig["ORGANIZATION_ID"] = organizationID
+
+	telemetryConfigMap, err := co.operatorNSConfigMapLister.ConfigMaps(api.OpenShiftConsoleOperatorNamespace).Get(telemetry.TelemetryConfigMapName)
+	if err != nil {
+		return telemetryConfig, err
+	}
+
+	if len(telemetryConfigMap.Data) > 0 {
+		for k, v := range telemetryConfigMap.Data {
+			telemetryConfig[k] = v
+		}
+	}
```

Comment:
Maybe not for this PR since we need to get this merged, but I think we should create a new controller that persists all telemetry config to this config map, and then we can just consume it here.